tests/test_packages/test_skills/test_generic_seller/test_behaviours.py

```diff
@@ -0,0 +1,222 @@
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""This module contains the tests of the behaviour classes of the generic seller skill."""
+
+import logging
+from pathlib import Path
+from typing import cast
+from unittest.mock import patch
+
+from aea.test_tools.test_skill import BaseSkillTestCase
+
+from packages.fetchai.protocols.ledger_api.message import LedgerApiMessage
+from packages.fetchai.protocols.oef_search.message import OefSearchMessage
+from packages.fetchai.skills.generic_seller.behaviours import (
+    GenericServiceRegistrationBehaviour,
+    LEDGER_API_ADDRESS,
+)
+from packages.fetchai.skills.generic_seller.strategy import GenericStrategy
+
+from tests.conftest import ROOT_DIR
+
+
+class TestSkillBehaviour(BaseSkillTestCase):
+    """Test behaviours of generic seller."""
+
+    path_to_skill = Path(ROOT_DIR, "packages", "fetchai", "skills", "generic_seller")
+
+    @classmethod
+    def setup(cls):
+        """Setup the test class."""
+        super().setup()
+        cls.service_registration = cast(
+            GenericServiceRegistrationBehaviour,
+            cls._skill.skill_context.behaviours.service_registration,
+        )
+        cls.strategy = cast(GenericStrategy, cls._skill.skill_context.strategy)
+
+    def test_service_registration_behaviour_setup_is_ledger_tx(self):
+        """Test the setup method of the service_registration behaviour where is_ledger_tx is True."""
+        # setup
+        self.strategy._is_ledger_tx = True
+        mocked_description_1 = "some_description_1"
+        mocked_description_2 = "some_description_2"
+
+        # operation
+        with patch.object(
+            self.strategy,
+            "get_location_description",
+            return_value=mocked_description_1,
+        ):
+            with patch.object(
+                self.strategy,
+                "get_register_service_description",
+                return_value=mocked_description_2,
+            ):
+                with patch.object(
+                    self.service_registration.context.logger, "log"
+                ) as mock_logger:
+                    self.service_registration.setup()
+
+        # after
+        quantity = self.get_quantity_in_outbox()
+        assert (
+            quantity == 3
+        ), f"Invalid number of messages in outbox. Expected 3. Found {quantity}."
+
+        # message 1
+        has_attributes, error_str = self.message_has_attributes(
+            actual_message=self.get_message_from_outbox(),
+            message_type=LedgerApiMessage,
+            performative=LedgerApiMessage.Performative.GET_BALANCE,
```

Comment:
The following would help:

Taking out `XMessage.Performative` from `XMessage` class into its own independent class in the same module and renaming them as `XPerformative` (e.g. `FipaPerformative`). These can subclass a base Performative enum that resides in the same module as the base Message. The enum can then also have a property, which gives the XMessage class it is linked to. This way one can do:

```
performative.message_class
```

to access the message the performative belongs. This has several benefits, an immediate one being in the above function, you don't need to specify the message class. From a performative, its appropriate message_class is always accessible.

P.S. as a side note, it seems having inner classes in python is not a very encouraged practice, as opposed to other languages:
https://stackoverflow.com/a/2024583/1460594
https://stackoverflow.com/questions/719705/what-is-the-purpose-of-pythons-inner-classes/722175#722175